numpy/core/numeric.py

```diff
@@ -1455,6 +1459,93 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
+def _validate_axis(axis, ndim, argname):
+    if isinstance(axis, int) or getattr(axis, 'ndim', 1) == 0:
+        axis = [axis]
+    axis = list(axis)
```

Comment:
wrong line (phone). Is the above line not better with operator.index logic? It is a try except I admit, but the most general int accepting possible AFAIK.